Add deleteRow method

ilokhov opened the pull request:
This PR adds `deleteRow` method which allows completely removing a row from a dataset by row index.

This will be used by [the "Delete row" button in new map data step](https://github.com/datawrapper/plugin-simple-maps/pull/36).

There is already a [`filterRows` method in `Column`](https://github.com/datawrapper/chart-core/blob/7c02bdc6f1117c01167d1b1095fd461aba450ec0/lib/dw/dataset/column.js#L213-L226) which is similar. However it does not work for our purposes in map data step, since it relies on `origRows` which is a constant that is set once per instance. But we need to know the current structure of rows when deleting a row.

jakubdatawrapper commented on lib/dw/dataset/column.js:
```diff
@@ -225,6 +225,15 @@ export default function Column(name, rows, type) {
         },
 
+        deleteRow(i) {
+            if (!arguments.length) return;
```
I would remove this condition and just let the method crash if it's called without arguments (which is wrong and unsupported).

gka replied:
how about
```suggestion
            if (!arguments.length) throw new Error('deleteRow must be called with a row index');
```
then?

jakubdatawrapper replied:
You could add such a condition to any function that takes arguments. Like `Column.each()`, `Column.limitRows()`, `Dataset.deleteRow()` etc. I really think crashing is fine. I wouldn't be super defensive.

jakubdatawrapper commented on lib/dw/dataset/column.js:
```diff
@@ -225,6 +225,15 @@ export default function Column(name, rows, type) {
 
+        deleteRow(i) {
+            if (!arguments.length) return;
+            rows.splice(i, 1);
```
I think we should remove the row from `origRows` as well, otherwise `values()` with the `unfiltered` argument would return a different row than when called without the `unfiltered` argument, which might be unexpected.

jakubdatawrapper reviewed (changes requested):
Looks great. Thanks for the test!

I have two minor notes but other than that I think we're good to go.

ilokhov commented:
At first I wanted to go with @gka 's suggestion on throwing an error but then had a quick look at the code - as you mentioned @jakubdatawrapper, there are other similar scenarios where we don't throw errors, so I removed the check for consistency.

Also, well spotted about `origRows` @jakubdatawrapper, I implemented the change.

Please review again.

jakubdatawrapper reviewed (approved):
:rocket:
